flink-core/src/main/java/org/apache/flink/api/common/ExecutionConfig.java

```diff
@@ -220,6 +219,7 @@ public boolean isClosureCleanerEnabled() {
 	 */
 	@PublicEvolving
 	public ExecutionConfig setAutoWatermarkInterval(long interval) {
+		Preconditions.checkArgument(interval > 0, "The interval must be greater than 0.");
```

Comment:
interval >= 0

Reply:
Hi, I don't think this can be equal to 0.

In the case of equal to 0, it actually causes the watermark to remain at Long.MIN_VALUE, which is not what we want.